common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -424,6 +424,16 @@ def editor_saved(self, user, old_metadata, old_content):
         This should be fixed too.
         """
         metadata_was_changed_by_user = old_metadata != own_metadata(self)
+        # If metadata is not changed by user and html5_sub does not exist.
+        if self.sub and old_metadata.get('html5_sources', []) and not metadata_was_changed_by_user:
+            html5_ids = get_html5_ids(old_metadata['html5_sources'])
+            for subs_id in html5_ids:
+                try:
+                    Transcript.asset(self.location, subs_id)
+                except NotFoundError:
+                    metadata_was_changed_by_user = True
+                    break
```

Comment:
If any of transcripts does not exist than we does not need to further check other transcripts also we have to create that missing transcript as well using ```metadata_was_changed_by_user``` flag.